### PR TITLE
[stable/chronograf] Support OAuth (Google, Heroku, Github) and bump version

### PR DESCRIPTION
**Pull request by jackzampolin**

### CODE REVIEW

File: stable/chronograf/values.yaml [hunk 1]
```diff
@@ -29,10 +34,23 @@ resources:
...
+    kubernetes.io/ingress.class: "nginx"
+    secretName: my-tls-cert
+  # kubernetes.io/acme-tls: "true"
+
+## OAuth Settings for Github OAuth
+##
+oauth:
+  enabled: false
+  github:
+    client_id: f612alskdfc91cad5ad
```
Comment (viglesiasce) [inline stable/chronograf/values.yaml]: These shouldnt be set here

**Review by viglesiasce — CHANGES_REQUESTED** [review]
Looks like you left in your token and secret

File: stable/chronograf/values.yaml (hunk 1 above)
Comment (jackzampolin) [reply stable/chronograf/values.yaml]: That's actually a random string. Ok to leave?

Comment (viglesiasce) [reply stable/chronograf/values.yaml]: I would make that very clear by commenting this section out as they are not useful and ensuring that the value is something that denotes that it needs to be changed (ie `CHANGE_ME`)

File: stable/chronograf/values.yaml [hunk 2]
```diff
@@ -29,10 +34,23 @@ resources:
...
+## OPTIONALLY you can set .Values.ingress.secretName to set which secret to use    
 ## 
 ingress:
   enabled: false
   tls: false
   hostname: chronograf.foobar.com
   annotations:
-    kubernetes.io/ingress.class: "nginx"
\ No newline at end of file
+    kubernetes.io/ingress.class: "nginx"
```
Comment (viglesiasce) [inline stable/chronograf/values.yaml]: I would leave this annotation off by default so that the cluster default is used out of the box

**Review by viglesiasce — DISMISSED** [review]

**Comment by mgoodness** [comment]
@jackzampolin are you able to resolve this? I have additional review, but only if the PR is still active. Please let us know by 4 March, or we'll close.

**Comment by jackzampolin** [comment]
@mgoodness Sorry this fell through the cracks when getting back up to speed. I just addressed the last comment. This should also close out https://github.com/kubernetes/charts/pull/677

**Comment by jackzampolin** [comment]
I've added support for all of the different OAuth providers that are [currently supported](https://github.com/influxdata/chronograf/blob/master/docs/auth.md) by Chronograf.

I'm happy to make any other changes to this that are needed.

@viglesiasce

File: stable/chronograf/templates/ingress.yaml [hunk 3]
```diff
@@ -13,9 +13,14 @@ metadata:
...
-  - hosts:
-    - {{ .Values.ingress.hostname }}
-    secretName: {{ template "fullname" . }}-tls
+    - hosts:
+      - {{ .Values.ingress.hostname | quote }}
+{{- $eval := quote .Values.ingress.secretName }}
+{{- if eq (len $eval) 0 }}
+      secretName: {{ .Values.ingress.secretName }}
+{{- else }}
+      secretName: {{ template "fullname" . }}-tls
```
Comment (prydonius) [inline stable/chronograf/templates/ingress.yaml]: Could this be:

```
{{ .Values.ingress.secretName | default (printf "%s-tls" (include "fullname" .)) }}
```

**Review by prydonius — COMMENTED** [review]
code review lgtm, just a minor comment

File: stable/chronograf/templates/ingress.yaml (hunk 3 above)
Comment (jackzampolin) [reply stable/chronograf/templates/ingress.yaml]: Yes it could easily. And it will be!

**Comment by jackzampolin** [comment]
@prydonius Good catch! Its ready to go!

**Comment by jackzampolin** [comment]
@viglesiasce is this ready to go?

**Comment by mgoodness** [comment]
@k8s-bot ok to test

**Pull request merged**